crates/polars-io/src/parquet/write/batched_writer.rs

```diff
@@ -116,7 +121,17 @@ impl<W: Write> BatchedWriter<W> {
     /// Writes the footer of the parquet file. Returns the total size of the file.
     pub fn finish(&self) -> PolarsResult<u64> {
         let mut writer = self.writer.lock().unwrap();
-        let size = writer.end(None)?;
+
+        let key_value_metadata = self
```

Comment:
I think this is fine, but ideally, it would pass the metadata we write already so that it can be modified.